thinc/backends/_custom_kernels.py

```diff
@@ -303,8 +317,8 @@ def reduce_max(X, lengths, *, threads_per_block=128, num_blocks=128):
     _check_lengths(lengths, T)
 
     out_shape = (B, O)
-    maxes = cupy.zeros(out_shape, dtype=X.dtype)
-    which = cupy.zeros(out_shape, dtype="i")
+    maxes = _alloc(out_shape, dtype=X.dtype, zeros=False)
+    which = _alloc(out_shape, dtype="i", zeros=False)
```

Comment:
Are we sure about this one? `reduce_max` is retrieving values from `maxes` and `which`, no?

Reply:
Yes, `reduce_max` should fill all elements (like `maxout`), so `zeros=False` should be correct.

Reply:
Ok cool, thanks for double checking!